src/Component/Image.php

```diff
@@ -0,0 +1,44 @@
+<?php
+/**
+ * Class Image
+ *
+ * @package AMP
+ */
+
+namespace Amp\AmpWP\Component;
+
+use DOMElement;
+
+/**
+ * Class Image
+ *
+ * @internal
+ * @since 1.5.0
+ */
+class Image {
+
+	/**
+	 * The image node.
+	 *
+	 * @var DOMElement
+	 */
+	protected $image_node;
+
+	/**
+	 * Constructs the class.
+	 *
+	 * @param DOMElement $image_node The image node.
+	 */
+	public function __construct( DOMElement $image_node ) {
+		$this->image_node = $image_node;
```

Comment:
Should this throw an Exception if `$image_node` is not an `amp-img`, `img`, or `amp-anim`?

Reply:
That's a good idea. I'll make a commit for that.

Reply:
Also, it seems somewhat needless to have this class, as it is just a wrapper around `DOMElement`. Since non-image elements can also be used in a carousel, it would be more flexible to allow any DOM element actually.

So instead of having an `Image` class, could we just use `DOMElement`. And then for elements with captions, could this then be a subclass of `DOMElement` like `CaptionedElement` which takes a `$caption` arg in its constructor?

Reply:
Or rather, `CaptionedSlide`. I suppose it couldn't be a subclass because you're working with existing DOM elements. But it could be a container object that has the existing DOM element as a member?

Reply:
Sure, what do you think about https://github.com/ampproject/amp-wp/pull/3659/commits/12c6d2694589adce9e9293550f62a0a3eeb4e18f ?

`class Image` is deleted, but much of it is [moved to CaptionedSlide](https://github.com/ampproject/amp-wp/pull/3659/commits/12c6d2694589adce9e9293550f62a0a3eeb4e18f#diff-7d849810f36483c7db89977ff845c00fR25).

Maybe it's a slight improvement.

Reply:
The point of the `Image` class was to ensure that the `ImageList` only ever can accept images, not random elements.

This is one of the cornerstones of strong typing, that you actually have types to check against.

If our `ImageList` accepts any element, it is actually an `ElementList`, and it defeats the purpose.

Reply:
Ah, I just saw the comment about the carousel also accepting other elements.

But we should rename the `ImageList` then if it accepts non-images as well.

There's already a `DOMNodeList`, so we could indeed add `DOMElementList` just for typing purposes, it would improve typing and remove the need for these silly large docblocks stating a node is an element, just to get its attributes.